### PR TITLE
Catch JSON.parse errors when handling 'callApi' error response text

### PR DESCRIPTION
**Pull request by RyanWirth**
# Description

On Addition Elle, we saw the OCAPI backend return 404's/500's with (presumably) HTML response bodies. This caused the SSR server rendering to hang, so it responded with 502's after the 20 second timeout.

Here, we now catch JSON parse errors so that the SSR server will still respond with a rendered page.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Changes

- Wrapped `JSON.parse()` call in a try/catch block

# How to test this PR?

- Checkout the `additionelle` repo
- In this repo, run `npm link`
- In `additionelle/web`, run `npm link commercecloud-ocapi-client`
- In `additionelle/web/app/store/index.js`, change `getConnectorBasePath()` to return `"/page-not-found"`
   - This will cause the OCAPI requests to point at `/page-not-found`, which returns a 404 and includes an HTML response body
- In `additionelle/web`, run `npm run local`
- Navigate to https://localhost:3443/en/apparel-pants-flared
- **Verify** the SSR server renders the page (doesn't hang)

# Checklist:

- [x] My code follows the style guidelines of this project (`npm run lint`)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (`README.md` and `CHANGELOG.md`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`npm test`)

### CODE REVIEW

**Comment by johnboxall** [comment]
Non-blocking: Is it possible to write a test to demonstrate that this fixes the failure?

**Comment by RyanWirth** [comment]
> Non-blocking: Is it possible to write a test to demonstrate that this fixes the failure?

@johnboxall Yes -- I've extracted `callApi()`'s return Promise into its own function to make it easier to unit test.

File: src/ApiClient.js [hunk 1]
```diff
@@ -494,15 +494,32 @@ export default class ApiClient {
...
 
                     // Looks like there was an fault returned from the API
                     const hasErrorMessage = error.response && error.response.text
                     if (hasErrorMessage) {
-                        const fault = Fault.constructFromObject(JSON.parse(error.response.text).fault)
-                        reject(fault)
+                        try {
+                            const fault = Fault.constructFromObject(JSON.parse(error.response.text).fault)
+                            reject(fault)
+                        } catch (e) {
```
Comment (bendvc) [inline src/ApiClient.js]: Can you change `e` to `err`. Just a consistency thing with the rest of the library.

File: src/ApiClient.js [hunk 2]
```diff
@@ -494,15 +494,32 @@ export default class ApiClient {
...
+                            // If the error response text isn't valid JSON (eg. the API
```
Comment (bendvc) [inline src/ApiClient.js]: I don't hate large comments. But can be said simply like 'Reject immediately on parsing error.'

**Pull request merged**